nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -313,7 +324,47 @@ public static Stream<? extends AnnotationMirror> getTypeUseAnnotations(
     } else {
       // filter for annotations directly on the type
       return rawTypeAttributes.filter(
-          t -> NullabilityUtil.isDirectTypeUseAnnotation(t, symbol, config));
+          t ->
+              targetTypeMatches(symbol, t.position)
+                  && NullabilityUtil.isDirectTypeUseAnnotation(t, symbol, config));
+    }
+  }
+
+  // Adapted from Error Prone MoreAnnotations:
+  // https://github.com/google/error-prone/blob/5f71110374e63f3c35b661f538295fa15b5c1db2/check_api/src/main/java/com/google/errorprone/util/MoreAnnotations.java#L128
+  private static boolean targetTypeMatches(Symbol sym, TypeAnnotationPosition position) {
+    switch (sym.getKind()) {
+      case LOCAL_VARIABLE:
+        return position.type == TargetType.LOCAL_VARIABLE;
+      case FIELD:
+      // treated like a field
+      case ENUM_CONSTANT:
+        return position.type == TargetType.FIELD;
+      case CONSTRUCTOR:
+      case METHOD:
+        return position.type == TargetType.METHOD_RETURN;
+      case PARAMETER:
+        switch (position.type) {
+          case METHOD_FORMAL_PARAMETER:
```

Comment:
If we care only about one case, why is this a switch and not an `if`?

Reply:
dac889de3985f5b0614f71126560454134de99de